client.go

```diff
@@ -1011,3 +1031,42 @@ func (c *TunaSessionClient) IsSessClosed(sessKey string) bool {
 func (c *TunaSessionClient) SetTunaNode(node *types.Node) {
 	c.tunaNode = node
 }
+
+func (c *TunaSessionClient) getCachedPubAddrs(remoteAddr string) *PubAddrs {
+	c.RLock()
+	cachedAddr, ok := c.pubAddrs[remoteAddr]
+	c.RUnlock()
+	if !ok {
+		return nil
+	}
+
+	for i := 0; i < len(cachedAddr.Addrs); i++ {
```

Comment:
This is too slow. I would recommend we add another map addrConnCount map[remoteAddr]map[tcpAddr]count and maintain it the same way as `connCount`. When any count drops to 0 we remove `c.pubAddrs[remoteAddr]` from cache

Reply:
Yes, modified.